pkg/storegateway/indexheader/reader_pool_test.go

```diff
@@ -28,8 +28,11 @@ import (
 
 func TestReaderPool_NewBinaryReader(t *testing.T) {
 	tests := map[string]struct {
-		lazyReaderEnabled     bool
-		lazyReaderIdleTimeout time.Duration
+		lazyReaderEnabled         bool
+		lazyReaderIdleTimeout     time.Duration
+		eagerLoadReaderEnabled    bool
+		persistLazyLoadedHeaderFn func(blockId ulid.ULID, lazyLoadedSnapshotPath string)
```

Comment:
can this function be simplified to only accept the blockID and return the snapshot? The test can take care of persisting the returned snapshot in the correct path